<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>809</width>
    <height>649</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>HiCOM</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout_9">
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>100</horstretch>
        <verstretch>100</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="tab">
       <attribute name="title">
        <string>串口</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_10">
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_5" stretch="2,1,1">
          <item>
           <widget class="QGroupBox" name="groupBox">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="title">
             <string>串口设置</string>
            </property>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout">
               <item>
                <widget class="QLabel" name="label">
                 <property name="text">
                  <string>端口</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="serialCb"/>
               </item>
             </layout>
            </item>
            <item>
             <layout class="QHBoxLayout" name="horizontalLayout_2">
              <item>
               <widget class="QLabel" name="label_2">
                <property name="text">
                  <string>波特率</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="baundrateCb">
                 <property name="editable">
                  <bool>true</bool>
                 </property>
                 <property name="currentIndex">
                  <number>6</number>
                 </property>
                 <item>
                  <property name="text">
                   <string>1200</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>2400</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>4800</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>9600</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>19200</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>38400</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>115200</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_3">
               <item>
                <widget class="QLabel" name="label_3">
                 <property name="text">
                  <string>数据位</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="databitCb">
                 <property name="currentIndex">
                  <number>3</number>
                 </property>
                 <item>
                  <property name="text">
                   <string>5</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>6</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>7</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>8</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_4">
               <item>
                <widget class="QLabel" name="label_4">
                 <property name="text">
                  <string>停止位</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="stopbitCb">
                 <item>
                  <property name="text">
                   <string>1</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>1.5</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>2</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_5">
               <item>
                <widget class="QLabel" name="label_5">
                 <property name="text">
                  <string>校验位</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="checkbitCb">
                 <property name="currentText">
                  <string>None</string>
                 </property>
                 <property name="currentIndex">
                  <number>0</number>
                 </property>
                 <item>
                  <property name="text">
                   <string>None</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>奇校验</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>偶校验</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <layout class="QHBoxLayout" name="horizontalLayout_flow">
               <item>
                <widget class="QLabel" name="label_flow">
                 <property name="text">
                  <string>流控</string>
                 </property>
                </widget>
               </item>
               <item>
                <widget class="QComboBox" name="flowCtrlCb">
                 <property name="currentIndex">
                  <number>0</number>
                 </property>
                 <item>
                  <property name="text">
                   <string>None</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>RTS/CTS</string>
                  </property>
                 </item>
                 <item>
                  <property name="text">
                   <string>XON/XOFF</string>
                  </property>
                 </item>
                </widget>
               </item>
              </layout>
             </item>
             <item>
              <widget class="QPushButton" name="btnSerialCheck">
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>30</height>
                </size>
               </property>
               <property name="text">
                <string>保存日志</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QPushButton" name="openBt">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="minimumSize">
                <size>
                 <width>0</width>
                 <height>30</height>
                </size>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>100</height>
                </size>
               </property>
               <property name="text">
                <string>打开串口</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_2">
            <property name="title">
             <string>接收设置</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_recv">
             <item row="0" column="0">
              <widget class="QPushButton" name="clearBt">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>50</height>
                </size>
               </property>
               <property name="text">
                <string>清空</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QCheckBox" name="chk_rev_hex">
               <property name="text">
                <string>HEX</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QCheckBox" name="chk_rev_ansi">
               <property name="text">
                <string>ANSI解析</string>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QCheckBox" name="chk_rev_time">
               <property name="text">
                <string>时间戳</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <spacer name="verticalSpacer_recv">
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="sizeHint" stdset="0">
                <size>
                 <width>20</width>
                 <height>20</height>
                </size>
               </property>
              </spacer>
             </item>
             <item row="2" column="1">
              <widget class="QCheckBox" name="chk_rev_line">
               <property name="text">
                <string>自动换行</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
          <item>
           <widget class="QGroupBox" name="groupBox_3">
            <property name="title">
             <string>发送设置</string>
            </property>
            <layout class="QGridLayout" name="gridLayout_send">
             <item row="0" column="0">
              <widget class="QPushButton" name="btnClearSend">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximumSize">
                <size>
                 <width>16777215</width>
                 <height>50</height>
                </size>
               </property>
               <property name="text">
                <string>清空</string>
               </property>
              </widget>
             </item>
             <item row="0" column="1">
              <widget class="QCheckBox" name="chk_send_hex">
               <property name="text">
                <string>HEX</string>
               </property>
              </widget>
             </item>
             <item row="1" column="0">
              <widget class="QSpinBox" name="txtSendMs">
               <property name="maximum">
                <number>360000</number>
               </property>
               <property name="value">
                <number>200</number>
               </property>
              </widget>
             </item>
             <item row="1" column="1">
              <widget class="QCheckBox" name="chkTimSend">
               <property name="text">
                <string>自动发送ms</string>
               </property>
              </widget>
             </item>
             <item row="2" column="0">
              <widget class="QCheckBox" name="chkDtrSend">
               <property name="text">
                <string>DTR</string>
               </property>
              </widget>
             </item>
             <item row="2" column="1">
              <widget class="QCheckBox" name="chk_send_line">
               <property name="text">
                <string>发送新行</string>
               </property>
              </widget>
             </item>
            </layout>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_6" stretch="5,1">
          <property name="spacing">
           <number>6</number>
          </property>
          <item>
           <widget class="QTextEdit" name="recvEdit"/>
          </item>
          <item>
           <layout class="QHBoxLayout" name="horizontalLayout_8">
            <item>
             <widget class="QTextEdit" name="sendEdit"/>
            </item>
            <item>
             <widget class="QPushButton" name="sendBt">
              <property name="sizePolicy">
               <sizepolicy hsizetype="Minimum" vsizetype="Expanding">
                <horstretch>0</horstretch>
                <verstretch>0</verstretch>
               </sizepolicy>
              </property>
              <property name="baseSize">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
              <property name="text">
               <string>发送</string>
              </property>
              <property name="autoDefault">
               <bool>false</bool>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_waveform">
       <attribute name="title">
        <string>波形</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_wave">
        <item>
         <widget class="QLabel" name="wavePlaceholder">
          <property name="text">
           <string>波形将在运行时显示</string>
          </property>
          <property name="alignment">
           <set>Qt::AlignmentFlag::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="tab_3D">
       <attribute name="title">
        <string>3D</string>
       </attribute>
      </widget>
      <widget class="QWidget" name="tab_2">
       <attribute name="title">
        <string>关于</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_7">
        <item alignment="Qt::AlignmentFlag::AlignHCenter">
         <widget class="QLabel" name="label_6">
          <property name="font">
           <font>
            <pointsize>20</pointsize>
           </font>
          </property>
          <property name="text">
           <string>基于Qt与C++的自用串口助手HiCOM</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignmentFlag::AlignHCenter">
         <widget class="QLabel" name="label_8">
          <property name="text">
           <string>版本：1.0</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignmentFlag::AlignHCenter">
         <widget class="QLabel" name="label_7">
          <property name="text">
           <string>作者：Hayden</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QTextBrowser" name="textBrowser">
          <property name="html">
           <string>&lt;!DOCTYPE HTML PUBLIC &quot;-//W3C//DTD HTML 4.0//EN&quot; &quot;http://www.w3.org/TR/REC-html40/strict.dtd&quot;&gt;
&lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;meta charset=&quot;utf-8&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
p, li { white-space: pre-wrap; }
hr { height: 1px; border-width: 0; }
li.unchecked::marker { content: &quot;\2610&quot;; }
li.checked::marker { content: &quot;\2612&quot;; }
&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Microsoft YaHei UI'; font-size:9pt; font-weight:400; font-style:normal;&quot;&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:16pt; font-weight:700;&quot;&gt;	对于电子工程师来说，串口助手是在单片机开发过程中最常用的工具，而现有的串口助手各种用着不爽，于是便有了此工具满足自己的需求，需要什么功能都可以通过修改程序完善&lt;/span&gt;&lt;span style=&quot; font-size:16pt;&quot;&gt;。&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:16pt; font-weight:700;&quot;&gt;邮箱:hayhi@qq.com&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:16pt; font-weight:700;&quot;&gt;B站:极客大脑&lt;/span&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px; font-size:16pt; font-weight:700;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;img src=&quot;:/image/logo128.ico&quot; /&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;Emoji表情复制站：&lt;a href=&quot;https://emoji8.com/zh-hans/&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#27bf73;&quot;&gt;Emoji表情大全可复制 - Emoji8&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;span style=&quot; font-size:14pt;&quot;&gt;开源仓库：&lt;/span&gt;&lt;a href=&quot;https://github.com/HaydenHu/HiCOM&quot;&gt;&lt;span style=&quot; text-decoration: underline; color:#27bf73;&quot;&gt;https://github.com/HaydenHu/HiCOM&lt;/span&gt;&lt;/a&gt;&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="openExternalLinks">
           <bool>true</bool>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>809</width>
     <height>20</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
